modules/sdk-coin-ada/src/lib/transaction.ts

```diff
@@ -152,6 +206,21 @@ export class Transaction extends BaseTransaction {
       this._transaction = txn;
       this._id = Buffer.from(CardanoWasm.hash_transaction(txn.body()).to_bytes()).toString('hex');
       this._type = TransactionType.Send;
+      if (this._transaction.body().certs()) {
+        for (let i = 0; i < this._transaction.body().certs()!.len(); i++) {
+          const cert = this._transaction.body().certs()!.get(i);
+          if (cert.as_stake_registration() !== undefined) {
+            this._type = TransactionType.StakingActivate;
+          }
+          if (cert.as_stake_deregistration() !== undefined) {
+            this._type = TransactionType.StakingDeactivate;
+          }
+        }
+      }
+      if (this._transaction.body().withdrawals()) {
```

Comment:
Should this be an `else` or is it possible that the `certs` loop will be overwritten by this `if` statement?

Reply:
This certs loop should be overwritten here if this is the case!